include/unifex/reduce_stream.hpp

```diff
@@ -309,22 +315,23 @@ struct _sender<StreamSender, State, ReducerFunc>::type {
 
   template <typename Receiver>
   using operation = operation<StreamSender, State, ReducerFunc, Receiver>;
-
   template <typename Receiver>
-  operation<Receiver> connect(Receiver&& receiver) && {
-    return operation<Receiver>{
-        (StreamSender &&) stream_,
-        (State &&) initialState_,
-        (ReducerFunc &&) reducer_,
-        (Receiver &&) receiver};
-  }
-
+  using next_receiver = next_receiver<StreamSender, State, ReducerFunc, Receiver>;
```

Comment:
I thought it was ill-formed to use a name within class scope and have that name later change to refer to something else. Is this the one exception, where the only prior use of a name is in the RHS of a type-alias of the same name?

Reply:
I don't _think_ it's ill-formed, but I know some compilers warn, so I'll rename it.